### ISSUE TITLE
Module and directory naming convention

### ISSUE THREAD

**Issue by garthk** [opentelemetry-beam/opentelemetry_plug#2]
- `Opentelemetry` or `OpenTelemetry`?
- `OpenTelemetrySomething` or `OpenTelemetry.Something`? I was initially surprised at the former but I like how it avoids `alias` clashing with the `Something`
- `opentelemetry/something` or `open_telemetry/something` or `open_telemetry_something` or `opentelemetry_something`?

Asking because I've already been inconsistent in my `opentelemetry_honeycomb` and this project is doing something else again. :)

**Comment by garthk** [comment]
@tsloughter?

**Comment by garthk** [comment]
I'm going to follow the convention of `opentelemetry_api` having `OpenTelemetry.Span` in `lib/open_telemetry/span.ex`. If that needs a PR, so be it. :)

**Comment by tsloughter** [comment]
I did `lib/open_telemetry/` since `OpenTelemetry` is how the project name is written, maybe was the wrong way to go. 

Whether this is `OpenTelemetry.Plug` or `OpenTelemetryPlug` is partly an idiomatic Elixir question, right? But then also a question about the "namespacing"? I think its fine to use `OpenTelemetry.Plug` and we should be consistent so would suggest updating. But I should not be deciding what is idiomatic Elixir :)

Hell, for elixir integrations we could have a single app and do `OpenTelemetryContrib.Plug`, `OpenTelemetryContrib.Honeycomb`, etc?

I guess also `OpenTelemetryContrib.Plug` could be used without all of them being in the same application.

**Comment by garthk** [comment]
Yeah, I get a little uncomfortable with the last word in a module matching the last word in another module eg. `Plug` and `OpenTelemetry.Plug`, but that ship has sailed. Popular projects are rife with it.

 Strikes me you could have a `contrib` directory in the main repo if everyone was happy with the CLA, else sure an `OpenTelemetry.Contrib` with `pkg/opentelemetry_plug/lib/open_telemetry/contrib/plug.ex` defining a module `OpenTelemetry.Contrib.Plug` for the `Plug` integration.

**Comment by garthk** [comment]
Oh, bit of a corner: if I'm not creating a directory under `lib` for `lib/open_telemetry/plug.ex`, I'm likely to stick with `lib/opentelemetry_plug.ex`.
